src/activities/content/ContentModuleEntity.js

```diff
@@ -76,6 +76,23 @@ export class ContentModuleEntity extends Entity {
 		await performSirenAction(this._token, action, fields);
 	}
 
+	/**
+	 * Deletes the module
+	 */
+	async deleteModule() {
+		if (!this._entity) {
+			return;
+		}
+		const action = this._entity.getActionByName(Actions.content.deleteModule);
+		if (!action) {
+			return;
+		}
+
+		await performSirenAction(this._token, action).then(() => {
+			this.dispose();
```

Comment:
`this.dispose` goes [here](https://github.com/BrightspaceHypermediaComponents/siren-sdk/blob/master/src/es6/Entity.js#L44). Similar pattern to what assignments is doing on delete. According to the function it deletes the callbacks and listeners stored in the entity store.

Reply:
interesting to use a `.then` if we're already using an `asyc` function, imo.
> Code after each await expression can be thought of as existing in a .then callback. In this way a promise chain is progressively constructed with each reentrant step through the function. The return value forms the final link in the chain.

would doing 
```
await performSirenAction(this._token, action);
this.dispose();
```
not be the same?

Reply:
That is correct, I believe that would do the same thing. I'll test that out.

Reply:
Yeah, this works as expected and is cleaner. Have adjusted accordingly 👍